jenkins/Jenkinsfile

```diff
@@ -0,0 +1,268 @@
+def LinkedHashMap<String,String> getVersionMap(ArrayList<String> imageNames, String configFile) {
+  def LinkedHashMap<String,String> versionMap = [:]
+
+  //for example, if $image is 'welder-server', this line would isolate 'a4f0e04' from the line below in a $configFile
+  // welderDockerImage = "us.gcr.io/broad-dsp-gcr-public/welder-server:a4f0e04"
+  for (String image: imageNames) {
+    def version = sh(
+      script: "cat $configFile | grep $image | tr ':' '\n' | tail -1 | tr -d '\"' ",
+      returnStdout: true
+    ).trim()
+
+    versionMap[image] = version
+  }
+
+  return versionMap
+}
+
+pipeline {
+  agent { label 'node210' } 
+
+  options { 
+    //we don't want to be building more than one PR's changes at once, we will queue them instead
+    disableConcurrentBuilds() 
+  }
+
+  triggers {
+    // leaving the string empty causes it to trigger off pushes/PRs to the specified branch
+    pollSCM("")
+    //dataproc images expire every 60 days
+    cron("0 0 */59 * *")
+  }
+
+  parameters {
+    string(name: "IMAGE_BASE_NAME", defaultValue: "custom-leo-image",
+          description: "The name to give the custom dataproc image. This will be appended with an hypen and the short version of the commit hash (first 7 characters).")
+    booleanParam(name: "useCustomImages", defaultValue: false)
+    string(name: "GOOGLE_PROJECT", defaultValue: "broad-dsp-gcr-public",
+          description: "The google project to use")
+    string(name: "BUCKET_NAME", defaultValue: "gs://leo-dataproc-image",
+          description: "The bucket where the image will be stored")
+    string(name: "terra_jupyter_base", defaultValue: "0.0.1")
+    string(name: "terra_jupyter_hail", defaultValue: "0.0.1",
+          description: "Is not currently baked into the image, but is here for future functionality")
+    string(name: "terra_jupyter_r", defaultValue: "0.0.3")
+    string(name: "terra_jupyter_python", defaultValue: "0.0.1")
+    string(name: "terra_jupyter_bioconductor", defaultValue: "0.0.2",
+          description: "Is not currently baked into the image, but is here for future functionality")
+    string(name: "leonardo_jupyter", defaultValue: "dev",
+          description: "The tag of the jupyter image to pull in docker image build. Used in the prepare script.")
+    string(name: "welder_server", defaultValue: "latest",
+          description: "The tag of the welder image to pull in docker image build. Used in the prepare script.")
+    string(name: "openidc_proxy", defaultValue: "2.3.1_2",
+          description:"The tag of the openidc proxy image to pull in docker image build. Used in the prepare script.")
+  }
+
+  stages {
+    stage('Terra docker image scan') {
+      steps {
+        git credentialsId: 'jenkins-ssh-github', url: 'git@github.com:DataBiosphere/terra-docker.git', branch: 'master'
+
+        script {
+          def temp = readFile("jenkins/imageDirs.txt").trim().split("\n")
```

Comment:
Where does `imageDirs.txt` come from? Does that need to be committed?

Reply:
it lives in the terra-docker repo